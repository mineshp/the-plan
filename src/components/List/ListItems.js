import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import ListItemRow from './ListItemRow';

const ListItems = ({ items, handleChange }) => {
    const listItemRows = [];
    items.map((listRowObj) => (
        listItemRows.push(
            <ListItemRow
                key={listRowObj.rowId}
                itemRow={listRowObj}
                handleChange={handleChange}
            />
        )
    ));

    return (
        <Table.Body>
            {listItemRows}
        </Table.Body>
    );
};

ListItems.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({})
    ).isRequired,
    handleChange: PropTypes.func.isRequired
};

export default ListItems;
