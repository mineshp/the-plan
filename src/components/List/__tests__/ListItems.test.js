import React from 'react';
import renderer from 'react-test-renderer';
import ListItems from '../ListItems';

const mockItems = [
    {
        columns: [
            {
                columnId: '001',
                columnValue: 'Hola',
                columnPosition: 1,
                columnName: 'Spanish'
            },
            {
                columnId: '002',
                columnValue: 'Hello',
                columnPosition: 1,
                columnName: 'English'
            },
        ],
        id: '001'
    },
    {
        columns: [
            {
                columnId: '003',
                columnValue: 'Adios',
                columnPosition: 1,
                columnName: 'Spanish'
            },
            {
                columnId: '004',
                columnValue: 'Bye',
                columnPosition: 1,
                columnName: 'English'
            },
        ],
        id: '002'
    }
];

describe('ListHeadings', () => {
    it('renders headings correctly', () => {
        const tree = renderer.create(<ListItems items={mockItems} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
