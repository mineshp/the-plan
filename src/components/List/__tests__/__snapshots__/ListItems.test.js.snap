// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeadings renders headings correctly 1`] = `
<tbody
  className=""
>
  <tr
    className=""
  >
    <td
      className=""
    >
      Hola
    </td>
    <td
      className=""
    >
      Hello
    </td>
  </tr>
  <tr
    className=""
  >
    <td
      className=""
    >
      Adios
    </td>
    <td
      className=""
    >
      Bye
    </td>
  </tr>
</tbody>
`;
